woocommerce/google-listings-and-ads
# PMax Assets: Preprocessing for the assets form

### Changes proposed in this Pull Request:

This PR implements the preprocessing for the "Assets form" part of 📌 [Assets form and related components](https://github.com/woocommerce/google-listings-and-ads/issues/1787#fe-misc) in #1787.

In order to avoid the subsequent PR of the "Assets form" being too large, some are split into preprocessing:

- Move and rename *js/src/utils/paid-ads/validateForm* to *js/src/components/paid-ads/validateCampaign*
   - and a later PR will add the validateAssetGroup in the same directory
- Extract the campaign creation and editing forms out as `CampaignAssetsForm`
   - for adding more shared functions in a later PR
- Add `children` prop to `TextsEditor`
   - for showing error messages in a later PR
- Add `children` and `onChange` props to `ImagesSelector`
   - for showing error messages and getting updated image URLs in a later PR
- Add tests for `ImagesSelector`
- Add tests for the functions in `useCroppedImageSelector`


### Detailed test instructions:

1. Go to the campaign step of the campaign creation page
2. Check if the "Continue" or "Complete setup" button is disabled when the budget is <= 0
3. Check if the "Continue" or "Complete setup" button is disabled when the ads audience is empty
4. Repeat test step 2 for the campaign editing page
5. Repeat test steps 2 and 3 for the "Create your paid campaign" step of the ads setup flow (`/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsetup-ads`)
6. Repeat 1~3 steps for the "Complete your campaign" step of the onboarding flow (`/wp-admin/admin.php?page=wc-admin&path=%2Fgoogle%2Fsetup-mc`)

### Changelog entry
